src/modules/commander/state_machine_helper.cpp

```diff
@@ -1145,25 +1145,23 @@ void battery_failsafe(orb_advert_t *mavlink_log_pub, const vehicle_status_s &sta
 		// FALLTHROUGH
 		case LOW_BAT_ACTION::RETURN_OR_LAND:
 
-			// let us send the critical message even if already in RTL
-			if (TRANSITION_DENIED != main_state_transition(status, commander_state_s::MAIN_STATE_AUTO_RTL, status_flags,
-					internal_state)) {
-				mavlink_log_critical(mavlink_log_pub, "%s, Executing RTL", battery_critical);
+			if (status_flags.condition_global_position_valid && status_flags.condition_home_position_valid) {
+				internal_state->main_state = commander_state_s::MAIN_STATE_AUTO_RTL;
```

Comment:
Are we allowed to directly assign a new state by circumventing `main_state_transition()`? Perhaps because there would be no handling for the case where the state transition fails?

Reply:
If you look inside `main_state_transition()` it's not doing anything else. So basically I only moved the boolean checks outside. I didn't know how else to fix this case without too much changes. It still needs proper refactoring.
If you have a nicer fix for this, I'm all ears.

Thanks for the review!

Reply:
No I don't. I was just wondering what your rationale is. I think it looks fine and makes the code actually less obscure.